bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/NavigationImpl.java

```diff
@@ -171,18 +180,55 @@ public String getExportedType() {
             while (it.hasNext()) {
                 Page page = it.next();
                 int pageLevel = getLevel(page);
-                int level = pageLevel - navigationRoot.startLevel;
+                int level = pageLevel - navigationRoot.startLevel - 1;
                 List<NavigationItem> children = getItems(navigationRoot, page);
                 boolean isSelected = checkSelected(page);
-                if (skipNavigationRoot) {
-                    level = level - 1;
+                if (rootLevel == 0) {
+                    level = level + 1;
                 }
                 pages.add(new NavigationItemImpl(page, isSelected, request, level, children));
             }
         }
         return pages;
     }
 
+    private List<NavigationItem> getNavigationTree(NavigationRoot navigationRoot) {
+        List<NavigationItem> itemTree = new ArrayList<>();
```

Comment:
I'd refactor this block as follows:
```
if (rootLevel == 0) {
...
} else {
        Iterator<NavigationRoot> it = getRootItems(navigationRoot, rootLevel).iterator();
        while (it.hasNext()) {
            NavigationRoot item = it.next();
            itemTree.addAll(getItems(item, item.page));
        }
}
```

Reply:
That won't work as the `itemTree` needs to be passed at https://github.com/adobe/aem-core-wcm-components/pull/836/files#diff-0998f9ceef65d53968bc5519e4b9971fR202